src/test/java/it/pagopa/pn/cucumber/steps/recipient/RicezioneNotificheWebSteps.java

```diff
@@ -751,4 +764,230 @@ private static void waitState(Integer waitingStateCsv) {
+            }
+            Assertions.assertFalse(exists, "SERCQ FOUND");
+        });
+
+    }
+
+    @And("viene verificata l'assenza di indirizzi Pec per il comune {string}")
+    public void viewedPecPerUtentePerEnte(String pa) {
+        String senderId = getSenderIdPa(pa);
+        Assertions.assertDoesNotThrow(() -> {
+            List<LegalAndUnverifiedDigitalAddress> legalAddressByRecipient = this.iPnWebUserAttributesClient.getLegalAddressByRecipient();
+            boolean exists = false;
+            if (legalAddressByRecipient != null && !legalAddressByRecipient.isEmpty()) {
+                exists = legalAddressByRecipient.stream()
+                        .anyMatch(address -> LegalChannelType.PEC.equals(address.getChannelType()) && senderId.equals(address.getSenderId()));
+            }
+            Assertions.assertFalse(exists, "PEC FOUND");
+        });
+    }
+
+    //Come da SRS Abilitazione Domicilio Digitale, address è una stringa fissa "x-pagopa-pn-sercq:send-self:notification-already-delivered"
+    @And("viene attivato il servizio SERCQ SEND per recapito principale")
+    public void attivazioneSercqSend() {
+        postRecipientLegalAddressSercq("default", "x-pagopa-pn-sercq:send-self:notification-already-delivered");
+    }
+
+    //Come da SRS Abilitazione Domicilio Digitale, address è una stringa fissa "x-pagopa-pn-sercq:send-self:notification-already-delivered"
+    @And("viene attivato il servizio SERCQ SEND per il comune {string}")
+    public void attivazioneSercqPerEnteSpecifico(String pa) {
+        String senderIdPa = getSenderIdPa(pa);
+        postRecipientLegalAddressSercq(senderIdPa, "x-pagopa-pn-sercq:send-self:notification-already-delivered");
+    }
+
+    @And("viene inserito un recapito legale {string}")
+    public void nuovoRecapitoLegale(String pec) {
+        postRecipientLegalAddress("default", pec, null, true);
+    }
+
+    @And("viene controllato che siano presenti pec verificate inserite per il comune {string}")
+    public void waitedAndViewedPecDiPiattaformaDi(String pa) {
+        String senderId = getSenderIdPa(pa);
+        Assertions.assertDoesNotThrow(() -> {
+            try {
+                Thread.sleep(180000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Sleep was interrupted", e);
+            }
+
+            List<LegalAndUnverifiedDigitalAddress> legalAddressByRecipient = this.iPnWebUserAttributesClient.getLegalAddressByRecipient();
+            if (legalAddressByRecipient != null && !legalAddressByRecipient.isEmpty()) {
+                boolean exists = legalAddressByRecipient.stream()
+                        .anyMatch(address -> LegalChannelType.PEC.equals(address.getChannelType()) && senderId.equals(address.getSenderId()));
+
+                Assertions.assertTrue(exists, "PEC NOT FOUND");
```

Comment:
questa logica per il recupero dei legalAddress e il controllo dell'esistenza viene ripetuta in diversi punti, è possibile estrapolarla in un metodo dedicato e richiamarlo quando necessario?